### PR TITLE
Change of quote <-> base

### PR DESCRIPTION
**Pull request by CalabashSquash**
### Motivation
It was decided that what we were referring to as `base` was actually `quote` and vice-versa

### Changes
- Changes all references to `base` to `quote`, and all references of `quote` to `base`
- Delete `interfaces/IAccount.sol`
- Remove `liquidatorBase` from `LibLiquidation.liquidationBalanceChanges
- Fixed `LibLiquidation.liquidationBalanceChanges`.
    - Before, it was giving the opposite quoteChange if liquidatee was short.
    - It should always just be:
        - Liquidator takes liquidatee's quote
        - Liquidatee takes negative of their own quote (gets reduced)
    - Same with base

### CODE REVIEW

File: contracts/TracerPerpetualSwaps.sol [hunk 1]
```diff
@@ -388,10 +394,10 @@ contract TracerPerpetualSwaps is
             // Calc the difference in funding rates, remove price multiply factor
             int256 fundingDiff = currentGlobalRate - currentUserRate;
 
-            // base - (fundingDiff * quote
-            accountBalance.position.base =
-                accountBalance.position.base -
-                PRBMathSD59x18.mul(fundingDiff, accountBalance.position.quote);
+            // quote - (fundingDiff * base
```
Comment (CalabashSquash) [inline contracts/TracerPerpetualSwaps.sol]:
```suggestion
            // quote - (fundingDiff * base)
```

File: contracts/lib/LibBalances.sol [hunk 2]
```diff
@@ -39,13 +39,13 @@ library Balances {
         returns (uint256)
     {
         /* cast is safe due to semantics of `abs` */
-        return PRBMathUD60x18.mul(uint256(LibMath.abs(position.quote)), price);
+        return PRBMathUD60x18.mul(uint256(LibMath.abs(position.base)), price);
     }
 
     /**
-     * @notice Calculates the marign as base + quote * quote_price
+     * @notice Calculates the marign as quote + base * base_price
```
Comment (CalabashSquash) [inline contracts/lib/LibBalances.sol]:
```suggestion
     * @notice Calculates the margin as quote + base * base_price
```

**Review by greenbookwebb — COMMENTED** [review]
Looks good. Base = derivative, quote = stablecoin

File: contracts/TracerPerpetualSwaps.sol [hunk 3]
```diff
@@ -8,7 +8,6 @@ import {Types} from "./Interfaces/Types.sol";
 import "./lib/LibPerpetuals.sol";
 import "./Interfaces/IOracle.sol";
 import "./Interfaces/IInsurance.sol";
-import "./Interfaces/IAccount.sol";
```
Comment (greenbookwebb) [inline contracts/TracerPerpetualSwaps.sol:11]: why are we deleting this

Comment (CalabashSquash) [reply contracts/TracerPerpetualSwaps.sol:11]: Because it isn't used.

File: contracts/TracerPerpetualSwaps.sol [hunk 4]
```diff
@@ -91,17 +90,23 @@ contract TracerPerpetualSwaps is
 
     /**
      * @notice Allows a user to deposit into their margin account
-     * @dev this contract must be an approvexd spender of the markets base token on behalf of the depositer.
-     * @param amount The amount of base tokens to be deposited into the Tracer Market account. This amount
+     * @dev this contract must be an approvexd spender of the markets quote token on behalf of the depositer.
```
Comment (kumar-ish) [inline contracts/TracerPerpetualSwaps.sol]: May as well fix the typo "approvexd"

Comment (kumar-ish) [reply contracts/TracerPerpetualSwaps.sol]:
```suggestion
     * @dev this contract must be an approved spender of the markets quote token on behalf of the depositer.
```

File: contracts/lib/LibBalances.sol [hunk 5]
```diff
@@ -100,9 +100,9 @@ library Balances {
 
         uint256 liquidationGasCost = liquidationCost * 6;
 
-        uint256 minimumBase = notionalValue / maximumLeverage;
+        uint256 baseMinimumMargin = notionalValue / maximumLeverage;
```
Comment (raymogg) [inline contracts/lib/LibBalances.sol]: Don't know if I'm a fan `baseMinimumMargin`. I get that it is the base minimum amount (and not minimum margin of base). Maybe we could just call this minimumMargin?

Comment (CalabashSquash) [reply contracts/lib/LibBalances.sol]: What about `minimumMarginWithoutGasCost` or something? It's a long variable name but it's only used in 1 other line and aptly describes what it does.

File: contracts/Liquidation.sol [hunk 6]
```diff
@@ -440,18 +438,18 @@ contract Liquidation is ILiquidation, Ownable {
...
+                    insuranceBalance.position.quote <
                     amountWantedFromInsurance.toInt256()
                 ) {
                     // Still not enough
                     amountTakenFromInsurance = uint256(
-                        insuranceBalance.position.base
+                        insuranceBalance.position.quote
                     );
-                    // insuranceBalance.position.base = 0;
+                    // insuranceBalance.position.quote = 0;
```
Comment (CalabashSquash) [inline contracts/Liquidation.sol]:
```suggestion
```

File: contracts/lib/LibBalances.sol (hunk 5 above)
Comment (raymogg) [reply contracts/lib/LibBalances.sol]: yeah I like that @BenjaminPatch

**Review by raymogg — APPROVED** [review]

**Review by sporejack — APPROVED** [review]
LGTM!

**Pull request merged**